### ISSUE TITLE
BasicLTIUtil.signProperties() does not work for ContentItemSelection messages

### ISSUE THREAD

**Issue by ppettit** [IMSGlobal/basiclti-util-java#40]
Because `BasicLTIConstants` does not know about content item specific parameters like `data` or `content_items` and so they get turned into custom parameters by `BasicLTIUtil.cleanupProperties()`.

is the best way to fix this just to add all the parameters from the content item spec into BasicLTIContents or will that cause problems elsewhere? I can do a PR if this is acceptable or you can suggest another way of fixing.

**Comment by pfgray** [comment]
Yeah, I think they should be probably included in `BasicLTIConstants.validPropertyNames`. I don't think this will break anyone, since if they're creating content-item launches, then they're not using this method anyway... 
Reviewing http://www.imsglobal.org/specs/lticiv1p0/specification-3, here's the launch parameters that I think need adding:

Launch to provider:
`content_item_return_url`
`accept_media_types`
`accept_media_types`
`accept_presentation_document_targets`
`accept_unsigned`
`accept_multiple`
`accept_copy_advice`
`title`
`text`
`auto_create`
`data`



Launch back to consumer:
`content_items`
`data`
`lti_log`
`lti_errormsg`
`lti_errorlog`

Any I'm missing?

**Comment by ppettit** [comment]
that looks right to me.

i've since found the sakai/tsugi version https://github.com/sakaiproject/sakai/tree/master/basiclti/tsugi-util that seems to be more actively developed and already does what i need though, so for now i've switched to that.
